docs/user-guide/bestpractices.rst

```diff
@@ -16,4 +16,40 @@ Here are the methods in increasing order of compute required for them.
+To help avoid these issues and ensure efficient processing, here are some strategies for managing memory usage and mitigating OOM challenges.
+
+Utilize RMM Options
+~~~~~~~~~~~~~~~~~~~
+`RAPIDS Memory Manager (RMM) <https://github.com/rapidsai/rmm>`_ is a package that enables you to allocate device memory in a highly configurable way.
+The NeMo Curator team has found several of its features to be especially useful for fuzzy deduplication, notably the connected components step.
+Here are some features which can help optimize memory usage:
+
+* Enable asynchronous memory allocation: Use the ``--rmm-async`` flag to allow RMM to handle memory allocation more efficiently, by allocating and deallocating GPU memory asynchronously.
+* Set a memory release threshold: For example, ``--rmm-release-threshold 50GB`` can help prevent holding onto excess memory, releasing unused memory when a certain limit is reached. Please keep in mind that using this flag may degrade performance slightly as RMM is busy releasing the unused memory.
+
+You can set these flags while calling your Python script directly, for example:
+
+.. code-block:: bash
+
+  python your_script.py --rmm-async --rmm-release-threshold 50GB
+
+Fuzzy Deduplication Guidelines
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Fuzzy deduplication is one of the most computationally expensive algorithms within the NeMo Curator pipeline.
+Here are some suggestions for managing memory use during fuzzy deduplication:
+
+* Reduce bucket counts: During deduplication, the data is grouped into buckets to compare and identify near-duplicate documents. Increasing the number of buckets increases the probability that two documents within a given Jaccard similarity score are marked as duplicates. This leads to more accurate results as it reduces the number of false negatives. However, increasing the number of buckets also increases the memory requirements from the increased number of hashes it needs to store. Thus, it is important to find an optimal balance between memory usage and deduplication accuracy. You can experiment with this by using the ``num_buckets`` parameter when initializing your ``FuzzyDuplicatesConfig``.
+* Reduce buckets per shuffle: Because duplicates are still considered bucket by bucket, reducing the ``buckets_per_shuffle`` parameter in the ``FuzzyDuplicatesConfig`` does not affect accuracy. Instead, reducing the buckets per shuffle helps lower the amount of data being transferred between GPUs. However, using a lower ``buckets_per_shuffle`` will increase the time it takes to process the data.
+* Adjust files per partition: Processing large datasets in smaller chunks can help reduce the memory load. When reading data with ``DocumentDataset.read_json`` or ``DocumentDataset.read_parquet``, start with a smaller ``files_per_partition`` value and increase as needed.
```

Comment:
We can also make the recommendation to target no larger than `2GB` partitions for the time being. As a todo: Some of the suggestions made here: https://github.com/rapidsai/cudf/pull/16821 may also apply to us. We can link to it in the future if needed.

Reply:
Updating now and adding link.